### PR TITLE
feat(docs) Contribution guide added

### PR DESCRIPTION
**Pull request by valasek**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [x] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
An initial version of the Contributing guide without AE dev guides moved.

### CODE REVIEW

**Comment by valasek** [comment]
Updated broken links found by https://www.brokenlinkcheck.com/ and removed QColorPicker replacement by QColor from an Updade guide as there is no such component as QColor in V1.

**Comment by nothingismagick** [comment]
/vue-components/color-picker

// quasar.conf.js
```
return {
  framework: {
    components: [
      'QColor'
    ]
  }
}
```

**Comment by valasek** [comment]
Thx, QColorPicker upgrade steps returned to the Upgrade guide and link to QColor component fixed.

File: docs/src/pages/layout/grid/introduction-to-flexbox.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Quasar provides lots of CSS classes to help you build your UI easily with the he
 Flexbox (upon which Quasar Flex CSS classes are based on) module aims at providing a more efficient way to lay out, align and distribute space among items in a container, even when their size is unknown and/or dynamic (thus the word “flex”).
 
 ::: tip
-This page covers the basic theory of Quasar Flex CSS classes and prepares you for the in-depth pages on [Grid Row](/layout/grid/row), [Grid Column](/layout/grid/column) and [Grid Gutter](/layouts/grid/gutter).
+This page covers the basic theory of Quasar Flex CSS classes and prepares you for the in-depth pages on [Grid Row](/layout/grid/row), [Grid Column](/layout/grid/column) and [Grid Gutter](/layout/grid/gutter).
```
Comment (nothingismagick) [inline docs/src/pages/layout/grid/introduction-to-flexbox.md]: Why is this commit here? its not part of the scope of contribution docs. (still nice catch)

File: docs/src/pages/start/contribution-guide.md [hunk 2]
```diff
@@ -3,4 +3,276 @@ title: Contribution Guide
 desc: How to contribute to Quasar Framework.
 ---
 
-This part of the documentation is not yet available, but it will be soon. Sorry for any inconvenience.
+This guide covers ways in which you can become a part of the ongoing development of Quasar.
+
+But, before we begin, a first important note. Everyone interacting in Quasar is expected to follow the [Quasar code of conduct](https://github.com/quasarframework/quasar/blob/dev/.github/CODE_OF_CONDUCT.md).
```
Comment (nothingismagick) [inline docs/src/pages/start/contribution-guide.md]: an important note:

File: docs/src/pages/start/contribution-guide.md [hunk 3]
```diff
@@ -3,4 +3,276 @@ title: Contribution Guide
...
+Install the required dependencies.
+```bash
+$ cd quasar
+$ yarn # or npm install
+```
+
+#### Running documentation against your local repository
+
+```bash
+$ quasar dev
```
Comment (nothingismagick) [inline docs/src/pages/start/contribution-guide.md]: $ quasar dev --mode ssr

File: docs/src/pages/start/contribution-guide.md [hunk 4]
```diff
@@ -3,4 +3,276 @@ title: Contribution Guide
...
+
+An easy way to start to contribute to Quasar is to generalize the code you have created on your project and publish it as a Quasar App Extension. Follow this guide on how to [create a new extension](../app-extensions/development-guide/introduction).
+
+When done, submit a PR on [Quasar Awesome](https://github.com/quasarframework/quasar-awesome/blob/master/README.md#community-app-extensions), share your achievement via [Quasar Forum](https://forum.quasar-framework.org/category/15/v1-app-extensions). After the review we will [publish your extension](../app-extensions/discover).
+
+## Contributing to the Quasar Core Source Code
+
+As with any project, there are rules to contributing. Ours are written here, please read them carefully. After that, read the [Quasar code of conduct](https://github.com/quasarframework/quasar/blob/dev/.github/CODE_OF_CONDUCT.md) and you’ll be ready to contribute to Quasars’s core repositories.
+
+An article [Look at the source code](https://medium.com/quasar-framework/wip-look-at-the-source-code-please-1b905ea4906) will help you to get familiar with a Quasar code base.
```
Comment (nothingismagick) [inline docs/src/pages/start/contribution-guide.md]: @smolinari - is this up to date with 1.0?

File: docs/src/pages/start/contribution-guide.md [hunk 5]
```diff
@@ -3,4 +3,276 @@ title: Contribution Guide
...
+``` bash
+$ yarn # or: npm install
+```
+
+#### Commonly used NPM scripts
+
+``` bash
+# Start dev server with a demo app. This app has Quasar source code linked directly so any change will trigger HMR (Hot Module Reload) on the dev server.
+# There's a section for each feature where tests are made.
+$ yarn dev [theme] # or: npm run dev [theme]
```
Comment (nothingismagick) [inline docs/src/pages/start/contribution-guide.md]: yarn dev [theme] is still a thing?
@rstoenescu

**Review by nothingismagick — CHANGES_REQUESTED** [review]
Great job. Lets ship this as soon as Razvan and Scott have signed off and my comments are resolved.

File: docs/src/pages/start/contribution-guide.md (hunk 4 above)
Comment (smolinari) [reply docs/src/pages/start/contribution-guide.md]: It's using the older version. I'll update it.

Scott

**Comment by rstoenescu** [comment]
Excellent. Thanks for this. Will polish it a bit and deploy it.

File: docs/src/pages/start/contribution-guide.md (hunk 5 above)
Comment (realtimeappsolutions) [reply docs/src/pages/start/contribution-guide.md]: Only 2 quick comments ...

1. Does anyone regularly monitor StackOverflow? Even if someone does, should we be directing users there? 

2. The Forum is mentioned before the Discord chat group, yet it appears to me that the main developers hang out mainly in the Discord Chat. Should we place the Discord Chat above the Forum in this document.

Both things are sort of related to ... Wouldn't it be better concentrate our efforts in one single location?

File: docs/src/pages/start/contribution-guide.md (hunk 4 above)
Comment (smolinari) [reply docs/src/pages/start/contribution-guide.md]: Updated!!!

Scott

**Pull request merged**